crates/code_generation/src/accessor.rs

```diff
@@ -1,383 +1,310 @@
+            );
+            quote! {
+                CyclerInstance::#instance_name => #accessor_path,
+            }
+        });
+        quote! {
+            match instance {
+                #(#variants)*
+            }
         }
-        token_stream.append(TokenTree::Group(Group::new(
-            Delimiter::Brace,
-            token_stream_within_match,
-        )));
-        token_stream
     } else {
-        path_to_accessor_token_stream_with_cycler_instance(
-            prefix_token_stream,
-            path,
-            reference_type,
-            None,
-        )
+        path_to_accessor_token_stream_with_cycler_instance(prefix, path, reference_type, None)
     }
 }
 
 fn path_to_accessor_token_stream_with_cycler_instance(
-    prefix_token_stream: TokenStream,
-    path: &[PathSegment],
-    reference_type: ReferenceType,
-    cycler_instance: Option<&str>,
+    prefix: TokenStream,
+    path: &Path,
+    reference_type: ReferenceKind,
+    cycler_instance: Option<&InstanceName>,
 ) -> TokenStream {
-    let mut token_stream = TokenStream::default();
-    let mut token_stream_within_method = None;
-
-    let path_contains_optional = path.iter().any(|segment| segment.is_optional);
-    if !path_contains_optional {
-        token_stream.append(TokenTree::Punct(Punct::new('&', Spacing::Alone)));
-        if ReferenceType::Mutable == reference_type {
-            token_stream.append(TokenTree::Ident(format_ident!("mut")));
+    let segments = path.segments.iter().map(|segment| {
+        let field = match segment.is_variable {
+            true => format_ident!("{}", cycler_instance.unwrap().to_case(Case::Snake)),
+            false => format_ident!("{}", segment.name),
+        };
+        match segment.is_optional {
+            true => match reference_type {
+                ReferenceKind::Immutable => quote! { #field.as_ref()? },
+                ReferenceKind::Mutable => quote! { #field.as_mut()? },
+            },
+            false => quote! { #field },
         }
-    }
-
-    token_stream.extend(prefix_token_stream);
-
-    for (index, segment) in path.iter().enumerate() {
-        {
-            let token_stream = token_stream_within_method
-                .as_mut()
-                .unwrap_or(&mut token_stream);
-
-            token_stream.append(TokenTree::Punct(Punct::new('.', Spacing::Alone)));
-            match (segment.is_variable, cycler_instance) {
-                (true, Some(cycler_instance)) => {
-                    token_stream.append(TokenTree::Ident(format_ident!(
-                        "{}",
-                        cycler_instance.to_case(Case::Snake)
-                    )));
-                }
-                _ => {
-                    token_stream.append(TokenTree::Ident(format_ident!("{}", segment.name)));
-                }
-            }
+    });
+    if path.contains_optional() {
+        quote! {
+            (|| Some(#prefix . #(#segments).*))()
         }
```

Comment:
Why do you want to get rid of it?
What did you have in mind?

In my opinion this is the correct way to implement it, since it's the emitted code is much closer to what the user wrote. We only need to convert some options to references and restrict the "return scope" of the `?` operator.